cosmwasm/packages/wormhole-bindings/src/fake.rs

```diff
@@ -122,12 +118,18 @@ impl WormholeKeeper {
             "guardian set expired"
         );
 
+        let d = digest(data).context("failed to calculate digest for data")?;
         if let Some(g) = this.guardians.get(sig.index as usize) {
-            let s = ecdsa::Signature::try_from(&*sig.signature).unwrap();
-            g.verifying_key()
-                .verify(data, &s)
-                .map(|()| Empty {})
-                .map_err(From::from)
+            let s = recoverable::Signature::try_from(&sig.signature[..])
+                .context("failed to decode signature")?;
+            let verifying_key = s
+                .recover_verify_key_from_digest_bytes(&d.secp256k_hash.into())
```

Comment:
why did this become more complex?

Reply:
Going to mark this resolved so that github will let me merge but please let me know if you'd like me to make any changes to this in a follow-up PR.